### PR TITLE
AISDK-202: Add Speaker Names option

### PR DESCRIPTION
**Pull request by beaudrychase**

### CODE REVIEW

File: src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java
Comment (amikofalvy) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:129]: Can you defer this explanation to the documentation site? I don't want to replicate code that we will need to update if we decide to add this to Ai transcription as well.

File: src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java
Comment (amikofalvy) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:536]: ditto as previous comment. I think we should keep optional in the description, but defer the explanation of human usage only to the API docs site. I don't feel too strongly on this point, so please push back if you think otherwise.

**Review by amikofalvy — APPROVED** [review]

**Review by amikofalvy — CHANGES_REQUESTED** [review]
Requesting changes for unit tests.

File: src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java
Comment (beaudrychase) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:129]: I guess I didn't consider this because I was just following the established pattern. Are you suggesting to delete the line saying that it's only for 'human' entirely and adding a link to the option in our documentation. I'm hesitant to add links because of the potential for broken links.

Comment (amikofalvy) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:129]: Yeah, up to you. I was just giving another opinion to consider for our docs.
I was trying to optimize for least number of SDK updates to release/ or change how a feature works

Comment (beaudrychase) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:129]: I guess now I'm wondering if there should be anything at all. Should we just delete the 'only for human' lines entirely and not replace them with anything? We don't mention similar restrictions on foreign languages or hipaa

Comment (amikofalvy) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:129]: You could continue with what you have and we can discuss it async at our next office hours

Comment (beaudrychase) [reply src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java:129]: I'll bring it up at OH

File: README.md
Comment (kbridbur) [inline README.md]:
```suggestion
speaker.setDisplayName('Steve');
```

File: src/test/java/ai/rev/speechtotext/unit/RevAiJobTest.java
Comment (kbridbur) [inline src/test/java/ai/rev/speechtotext/unit/RevAiJobTest.java]: I'm not sure this needs a separate unit test. The SDK has no knowledge or validation for human vs non human options. We can just keep it to one test where we set all options

Comment (beaudrychase) [reply src/test/java/ai/rev/speechtotext/unit/RevAiJobTest.java]: that's a good point, I'll update the tests

**Review by kbridbur — CHANGES_REQUESTED** [review]
SpeakerInfo object needs to be added to the transcript object for this info to actually be used

File: README.md
Comment (kbridbur) [inline README.md]: Middle and last name too please

File: src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java
Comment (kbridbur) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/RevAiJobOptions.java]:
```suggestion
  public List<SpeakerName> getSpeakerNames() {
```

File: src/main/java/ai/rev/speechtotext/models/asynchronous/SpeakerInfo.java
Comment (kbridbur) [inline src/main/java/ai/rev/speechtotext/models/asynchronous/SpeakerInfo.java]:
```suggestion
 * A SpeakerInfo object represents the information related to the speaker in a transcript
```

**Review by kbridbur — APPROVED** [review]
Seems like you confused the two objects in the request. Other than that looks good. Please verify everything works before merging

**Pull request merged**
